include/seqan/bam_io/bam_index_bai.h

```diff
@@ -211,117 +351,36 @@ jumpToRegion(FormattedFile<Bam, Input, TSpec> & bamFile,
-                    if (linearMinOffset != 0u)
-                        break;
-                    for (unsigned j = 1; j < length(index._linearIndices[i]); ++j)
-                    {
-                        if (index._linearIndices[i][j] > linearMinOffset)
-                        {
-                            linearMinOffset = index._linearIndices[i][j];
-                            break;
-                        }
-                    }
-                    if (linearMinOffset != 0u)
-                        break;
-                }
-            }
-        }
-        else
-        {
-            linearMinOffset = back(index._linearIndices[refId]);
-        }
-    }
-    else
-    {
-        linearMinOffset = index._linearIndices[refId][windowIdx];
-    }
-
-    // Combine candidate bins and smallest required offset from linear index into candidate offset.
-    typedef std::set<uint64_t> TOffsetCandidates;
-    TOffsetCandidates offsetCandidates;
-    typedef typename Iterator<String<uint16_t>, Rooted>::Type TCandidateIter;
-    for (TCandidateIter it = begin(candidateBins, Rooted()); !atEnd(it); goNext(it))
-    {
-        typedef typename std::map<uint32_t, BaiBamIndexBinData_>::const_iterator TMapIter;
-        TMapIter mIt = index._binIndices[refId].find(*it);
-        if (mIt == index._binIndices[refId].end())
-            continue;  // Candidate is not in index!
-
-        typedef typename Iterator<String<Pair<uint64_t, uint64_t> > const, Rooted>::Type TBegEndIter;
-        for (TBegEndIter it2 = begin(mIt->second.chunkBegEnds, Rooted()); !atEnd(it2); goNext(it2))
-            if (it2->i2 >= linearMinOffset)
-                offsetCandidates.insert(it2->i1);
-    }
-
-    // Search through candidate offsets, find rightmost possible.
-    //
-    // Note that it is not necessarily the first.
-    //
-    // TODO(holtgrew): Can this be optimized similar to how bamtools does it?
-    typedef typename TOffsetCandidates::const_iterator TOffsetCandidateIter;
-    BamAlignmentRecord record;
-    for (TOffsetCandidateIter candIt = offsetCandidates.begin(); candIt != offsetCandidates.end(); ++candIt)
-    {
-        setPosition(bamFile, *candIt);
-
-        readRecord(record, bamFile);
-
-        // std::cerr << "record.beginPos == " << record.beginPos << "\n";
-        if (record.rID != refId)
-            continue;  // Wrong contig.
-
-        if (record.beginPos <= regionStart)
-        {
-            offset = *candIt;
-        }
-
-        if (record.beginPos >= regionEnd)
-            break;  // Cannot find overlapping any more.
-    }
-
-    if (offset == std::numeric_limits<uint64_t>::max())
-        return true; // Finding no overlapping alignment is not an error, hasAlignments is false.
+    // -------------------------------------------------------------------------
+    uint64_t linearMinOffset;
+    if (!_getMinFileOffset(linearMinOffset, index, refId, regionStart))
```

Comment:
I think the linear index uses 0-based index for the windows.
So to make it consistent with viewRecords, I would assume that regionStart is passed as 1-based position. 
That should be then here corrected.